syntax = "proto3";
package distance_server;
import "google/api/annotations.proto";
import "google/protobuf/empty.proto";

message ClientInfo {
    string info = 1;
}

message ServerInfo {
    string info = 1;
}

message Distance {
    int32 id = 1;
    int64 ts = 2;
    int64 elapsed = 3;
    int32 distance = 4;
}

service DistanceServer {

    rpc registerClient (ClientInfo) returns (ServerInfo) {
    }

    rpc getDistances (ClientInfo) returns (stream Distance) {
        option (google.api.http) = {
			post: "/v1/distances"
			body: "*"
		};
    }

    rpc getDistance (google.protobuf.Empty) returns (Distance) {
        option (google.api.http) = {
            get: "/v1/distance"
        };
    }

    rpc resetElapsed (google.protobuf.Empty) returns (google.protobuf.Empty) {
        option (google.api.http) = {
            get: "/v1/resetElapsed"
        };
    }
}


